### DISCUSSION TITLE
Updates when we use Carelink

### DISCUSSION THREAD

**Discussion by 808IUFan** [NightscoutFoundation/xDrip#3640] [Q&A] [answered]
I love having all apps updated but when I tried to update xDrip+ the last time, nothing worked because is appears that Carelink Follower does not work on all versions. Is there a way we can put in the details of the different apps downloads if it works with Carelink Follower? I am using version de39292-dev-2024.01.21. Is there a newer version of the app that works with Carelink? #3422 #3424

**Comment by Navid200** [comment]
We have a had a new Nightly with some modifications for CareLink.  I don't know if it  will address your issue or not.
But, it is very easy to try different versions of xDrip as there is no need tp uninstall before installing another version.
So, even if you update and it does not work, you can always install the previous version again.

Release notes: https://navid200.github.io/xDrip/docs/ReleaseNotes.html

**Comment by 808IUFan** [comment]
Same error. App not updated. Conflicts with existing package for the latest
nightly build xDrip-plus-20240918-aee9315.apk

**Reply by Navid200** [reply]
Can you post a screenshot of your classic status page?
This is where it is: https://navid200.github.io/xDrip/docs/xDrip-Version.html

**Comment by 808IUFan** [comment]
[image: Screenshot_20240923-103146.png]

**Reply by Navid200** [reply]
Please don't respond to the email in order to post even though the GitHub email tells you you can.

Please log into GitHub and edit the post you want to make and tap on the paper clip symbol, which will let you attach an image.

**Reply by 808IUFan** [reply]
![Screenshot_20240923-103146](https://github.com/user-attachments/assets/67ab2346-4961-4685-9055-c38f8664a424)

**Reply by Navid200** [reply]
That release is not from this repository.

You can go to the repository you got that from and ask for advice.

If you want to use the releases from this repository, you will need to backup everything first: https://navid200.github.io/xDrip/docs/Backup-Database.html
Then, you will need to uninstall xDrip.
Then, you will be able to install xDrip from this repository.
Then, you can restore your backup.

**Reply by 808IUFan** [reply]
I tried that and Carelink Follower was not an available source. So I went back to the release I am currently on.

**Reply by Navid200** [reply]
I can see CareLink follower under Hardware Data Source in my xDrip, which is from this repository.

**Comment by 808IUFan** [comment]
![Screenshot_20240923-110105](https://github.com/user-attachments/assets/8a0686ea-3ea8-4ae4-a817-133494972d57)

**Reply by Navid200** [reply]
That's the source wizard.
It is needed for setting up something like a Dexcom collector that requires changing multiple settings.  And it only works right after you install xDrip for the first time.  I don't believe you can use it to change to a different data source. 


Please just go to Hardware Data source and choose CareLink: 
https://navid200.github.io/xDrip/docs/HardwareDataSource.html

**Reply by 808IUFan** [reply]
Ok so now that I explore more, if you click settings and change Nightscout or Dexcom, Carelink does appear and it imported the settings, but when you open the app, it immediately crashes. The widge works fine, but when you click the widget and open to full screen it immediately crashes and after several attempts, android asks if you want the close the app because it's repeatedly faulting.

**Reply by Navid200** [reply]
Please uninstall xDrip again.
Install it.  Don't restore the backup. 
You must be able to open the app, right?
Now, go to hardware data source and choose CareLink.  
It doesn't crash, does it?
Now, change the CareLink settings to get readings.  Does it work and not crash?

I suspect there may be a setting coming from the other repository that causes the crash.  But, I cannot be sure.

Is there anything in the previous install you absolutely need?

**Comment by 808IUFan** [comment, marked as answer]
I fixed it by restarting the phone. Maybe there was some garbage from the incompatible version. Thanks for your help.

**Reply by Navid200** [reply]
You're welcome

**Comment by 808IUFan** [comment]
Hey there. You helped me before I think when Medtronic had an issue. My
connection to Carelink in xDrip is down again but unlike the last issue
which was Medtronic's, their app is working fine. I get this when I try to
log in to carelink from xDrip. Their app works but when I try to test
logging into Carelink on my phone in Chrome, it also refuses. I put in my
U/P and hit login and it goes right back to the login page. This is exactly
what it did when they had an issue. I called them today and tried to get
them to acknowledge they had done something but of course since their app
works, they say it's you.

Carelink Follower is you or Medtronic? Who manages that? Can you provide
any help?
[image: Screenshot_20241115-165019.png]

**Reply by Navid200** [reply]
You should know that xDrip is not an official app.  Therefore, the last thing you should do is to contact a manufacturer and complain to them about a problem you are experiencing with xDrip.  It could result in the manufacturer limiting access, which will hurt not only you, but everyone else who uses xDrip.

Anyway, you should update to our latest Nightly release, which is here: https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.11.17

**Comment by 808IUFan** [comment]
Yeah I have done that. I have the latest release. And yes I know it's not
official but their app is so bad and I am used to xdrip.

I asked if they were aware of any issues with their service and they said
no. So I only asked if they knew anything about Carelink Follower and they
said no. I also told them I can not log into Carelink on my phone using
Chrome and told them that happened the last time they had an issue. Their
response was that our services are working. It took them 2 weeks the last
time to realize it was their issue and they had to update their app.

So are you aware of any issues on your end? I can't even get to the
Carelink website inside xDrip and on the Chrome Browser on my phone in
Desktop Mode, the login page goes right back to blank Username and Password
after I put those in. This started around 1:57pm last Friday Nov 15th.

**Reply by Navid200** [reply]
@benceszasz

**Comment by 808IUFan** [comment]
By the way, Nightscout works but it's inferior also and you have to pay.
Using Nightscout Tidepool also uploads.

**Comment by benceszasz** [comment]
@808IUFan 
Which version do you use?
Are you in US or outside US?

**Reply by 808IUFan** [reply]
I fixed it. Thanks

**Comment by benceszasz** [comment]
@Navid200 thread can be closed, issue was solved
